components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java

```diff
@@ -1074,4 +1096,70 @@ public synchronized boolean sendSiteCommand(String command) throws GenericFileOp
         // is not implemented
         return true;
     }
+
+    /*
+     * adapted from com.jcraft.jsch.Util.createSocket(String, int, int)
+     *
+     * added possibility to specify the address of the local network interface, against the
+     * connection should bind
+     */
+    static Socket createSocketUtil(String host, int port, String bindAddress, int timeout) {
+        Socket socket=null;
+        if(timeout==0){
+          try{
+            socket=bindAddress == null ? new Socket(host, port) : new Socket(InetAddress.getByName(host), port, InetAddress.getByName(bindAddress), 0);
+            return socket;
+          }
+          catch(Exception e){
+            String message=e.toString();
+            if(e instanceof Throwable)
+              throw new RuntimeException(message, (Throwable)e);
+            throw new RuntimeException(message);
+          }
+        }
+        final String _host=host;
+        final int _port=port;
+        final Socket[] sockp=new Socket[1];
+        final Exception[] ee=new Exception[1];
+        String message="";
+        Thread tmp=new Thread(new Runnable(){
```

Comment:
why new runnable needed? can you please elaborate on it?

Reply:
Ohh i see, i found a similar piece of this. I think this is public static method, no copy paste needed i guess. it may be imported. i did not fully check..

Reply:
I wanted to stay as close as possible to the original socket creation implementation. Therefore I copied the mentioned Util method (which has package protected access, btw) and added the possibility to specify the bind address to it (the original method does not support specifying the bind address). I've also added a comment on the method with a link to the original method.

The runnable here is used to implement the timeout behavior.